lib/etcdctl.py

```diff
@@ -90,10 +90,15 @@ def member_list(self, leader_address=None):
         # Expect output like this:
         # 4f24ee16c889f6c1: name=etcd20 peerURLs=https://10.113.96.197:2380 clientURLs=https://10.113.96.197:2379  # noqa
         # edc04bb81479d7e8: name=etcd21 peerURLs=https://10.113.96.243:2380 clientURLs=https://10.113.96.243:2379  # noqa
+        # edc0dsa81479d7e8[unstarted]: peerURLs=https://10.113.96.124:2380  # noqa
 
         for unit in raw_member_list:
             if '[unstarted]' in unit:
-                members['unstarted'] = {}
+                unit_guid = unit.split('[')[0]
```

Comment:
I'm trying to see if we can use `ETCDCTL_API=3 -w json`, so don't merge this quite yet. It's looking shaky though, so I'll probably retract this.

Reply:
Okay, this works:

```
$ ETCDCTL_CACERT=/var/snap/etcd/common/ca.crt ETCDCTL_CERT=/var/snap/etcd/common/client.crt ETCDCTL_KEY=/var/snap/etcd/common/client.key ETCDCTL_API=3 etcdctl --endpoints [https://10.205.42.38:2379] -w json member list
{"header":{"cluster_id":9581641538479738059,"member_id":7672478921058476181,"raft_term":2},"members":[{"ID":7672478921058476181,"name":"etcd0","peerURLs":["https://10.205.42.38:2380"],"clientURLs":["https://10.205.42.38:2379"]}]}
```

See https://github.com/coreos/etcd/issues/6915#issuecomment-264043931 for an explanation of why we need the different env vars.

Reply:
We have to support etcd 2.3 as well. I don't think we can use ETCDCTL_API=3 there, can we?

Reply:
Nope, it's no good with the etcdctl we ship with 2.3.8. We'll need to keep the manual string parsing until we can upgrade it.